app/src/main/java/com/github/houseorganizer/houseorganizer/house/CreateHouseholdActivity.java

```diff
@@ -36,23 +50,107 @@ protected void onCreate(Bundle savedInstanceState) {
 
         db = FirebaseFirestore.getInstance();
         mUserEmail = getIntent().getStringExtra("mUserEmail");
+
+        if(Geocoder.isPresent()){
+            geocoder = new Geocoder(this, Locale.getDefault());
+        }
+    }
+
+    private Location getCoordinatesFromAddress(String address) {
+        if(geocoder == null){
+            geocoder = new Geocoder(this, Locale.getDefault());
+            return null;
+        }else{
+            try {
+                Location location = new Location("");
+
+                List<Address> geoResults = geocoder.getFromLocationName(address, 1);
+                int count = 0;
+                while (geoResults.size()==0 && count < 10) {
+                    geoResults = geocoder.getFromLocationName(address, 1);
+                    ++count;
+                }
```

Comment:
does getFromLocationName need to run multiple times to find a correct location sometimes ? how is that ? I can imagine it can find more than one location per address but wouldnt it return all of them at once ?
Also this one looks deprecated : https://developer.android.com/reference/android/location/Geocoder#getFromLocationName(java.lang.String,%20int,%20double,%20double,%20double,%20double)
using the one with the listener instead shouldnt be too hard

Reply:
For the first part I found that sometimes for no reason it just returns null or an empty list, so this allows to ping it more than once to get an answer without having to push the button a second time.
For the second part it is deprecated for android 13 which is still in beta and the listener doesn't exist for the version we are using, even though I agree it would be way better

Reply:
Oh I see, that's my bad then, thanks for clarifying !